app/HutchManager/Models/DataSource.cs

```diff
@@ -1,15 +1,20 @@
+using HutchManager.Data.Entities;
+
 namespace HutchManager.Models;
 
 public class DataSource
 {
   public string Id { get; set; } = string.Empty;
   public DateTimeOffset LastCheckin { get; set; }
+  
+  public List<Agent>? Agents { get; set; }
```

Comment:
```suggestion
  public List<Agent> Agents { get; set; }
```

shouldn't be `null` - no items should just be an empty list, which is guaranteed by the constructor because the entity initialise `Agents` to `new()`.

Reply:
> shouldn't be `null` - no items should just be an empty list, which is guaranteed by the constructor because the entity initialise `Agents` to `new()`.

Makes sense, but removing this `List<Agent>?` brings back the uninitialised warning on `public DataSource()` so I will initialise it in the model as well like this:
```suggestion
  public List<Agent> Agents { get; set; } = new();
```